lib/redis_memo/memoize_method.rb

```diff
@@ -8,6 +8,27 @@
 require_relative 'util'
 
 module RedisMemo::MemoizeMethod
+  # Core entry method for using RedisMemo to cache a method's results. When a method is memoized, all
+  # calls to the method  will first check if the results exist in the RedisMemo cache before calling
+  # the original method.
+  #
+  # @example
+  #   class Post < ApplicationRecord
+  #     extend RedisMemo::MemoizeMethod
+  #     def display_title
+  #       "#{title} by #{author.display_name}"
+  #     end
+  #     memoize_method :display_title do |post|
+  #       depends_on Post.where(id: post.id)
+  #       depends_on User.where(id: post.author_id)
+  #     end
+  #   end
+  #
+  # @param method_name [String] The name of the method to memoize
+  # @param method_id [String] Optionally, a method_id that's used to tag APM traces of RedisMemo calls.
+  # @param options [Hash] Cache options to pass to RedisMemo. These values will override the global
+  #        cache options.
+  # @param depends_on [block] The method's dependency block
```

Comment:
This could also use a bit more details: How does one use the `depends_on` API? What does this method expect?